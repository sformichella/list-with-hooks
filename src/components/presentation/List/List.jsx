import React from 'react';
import PropTypes from 'prop-types';
import Character from '../Character/Character';

export default function List({ characters }) {
  return (
    <ul data-testid="character-list">
      {
        characters.map(character => {
          return <li key={character.id}><Character {...character}/></li>;
        })
      }
    </ul>
  );
}

List.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired
  }))
};
