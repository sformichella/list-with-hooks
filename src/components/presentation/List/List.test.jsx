import React from 'react';
import { render, cleanup } from '@testing-library/react';
import List from './List';

describe('List component', () => {
  afterEach(() => cleanup());
  it('renders List', () => {
    const { asFragment } = render(<List />);
    expect(asFragment()).toMatchSnapshot();
  });
});
