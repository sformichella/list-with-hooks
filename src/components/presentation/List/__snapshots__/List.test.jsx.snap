// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<DocumentFragment>
  <ul
    data-testid="character-list"
  >
    <li>
      <a
        href="/1"
      >
        Rick Sanchez
      </a>
      <img
        src="image.com"
      />
    </li>
  </ul>
</DocumentFragment>
`;
